src/shared/lib/ForegroundMessageHandler.tsx

```
@@ -25,6 +25,8 @@ function toSafeInternalPath(rawUrl?: string) {
 
 export function ForegroundMessageHandler() {
   const router = useRouter();
+  const routerRef = useRef(router);
+  routerRef.current = router;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/shared/lib/ForegroundMessageHandler.tsx:29:3
  27 |   const router = useRouter();
  28 |   const routerRef = useRef(router);
> 29 |   routerRef.current = router;
     |   ^^^^^^^^^^^^^^^^^ Cannot update ref during render
  30 |
  31 |   useEffect(() => {
  32 |     if (typeof window === 'undefined') return;

(react-hooks/refs)